core/src/dsnp/pseudo_relationship_identifier.rs

```diff
@@ -0,0 +1,79 @@
+use crate::dsnp::dsnp_types::DsnpPrid;
+use anyhow::{Error, Result};
+use crypto_box::{aead::AeadInPlace, Nonce, PublicKey, SecretKey};
+use hkdf::Hkdf;
+use sha2::Sha256;
+use std::ops::Deref;
+use x25519_dalek::StaticSecret;
+use xsalsa20poly1305::{KeyInit, XSalsa20Poly1305};
+use zeroize::Zeroizing;
+
+const PRI_CONTEXT: &[u8] = b"PRIdCtx0";
+struct PseudonymousRelationshipIdentifier;
+
+impl PseudonymousRelationshipIdentifier {
+	fn create(
+		a: u64,
+		b: u64,
+		a_secret_key: &SecretKey,
+		b_public_key: &PublicKey,
+	) -> Result<DsnpPrid> {
+		let id_b = b.to_le_bytes();
+
+		// calculate shared secret
+		let root_shared = StaticSecret::from(*a_secret_key.as_bytes())
+			.diffie_hellman(&x25519_dalek::PublicKey::from(*b_public_key.as_bytes()));
+
+		// derive a new key form pri context
+		let mut derived_key = Zeroizing::new([0u8; 32]);
+		let hk = Hkdf::<Sha256>::new(Some(&id_b[..]), root_shared.as_bytes());
+		hk.expand(PRI_CONTEXT, derived_key.as_mut_slice())
+			.map_err(|e| Error::msg(format!("key derivation error {:?}", e)))?;
+
+		// setting nonce with `b` for encryption
+		let mut nonce = [0u8; 24];
+		nonce[16..].copy_from_slice(&id_b[..]);
```

Comment:
I'm setting the last 8 bytes but I'm not totally sure if this is correct.
@wesbiggs Please take a look.

Reply:
Let's use the first 8 bytes (0-7). I will clarify this on the DSNP page.